### PR TITLE
80331 stryker report improvement

### PR DESCRIPTION
**Pull request by sakshi15525**
1. Improved Stryker score for API project.
2. Improved Stryker score for Common project.
3. Improved Stryker score for all 3 Web Job projects.

### CODE REVIEW

File: src/UKHO.ERPFacade.API/Helpers/SapMessageBuilder.cs [hunk 1]
```diff
@@ -33,7 +33,8 @@ public class SapMessageBuilder : ISapMessageBuilder
         private const string UnitSaleType = "unit";
         private const string EncCell = "ENC CELL";
         private const string AvcsUnit = "AVCS UNIT";
-
+        private const string RecDateFormat = "yyyyMMdd";
+        private const string RecTimeFormat = "hhmmss";
```
Comment (harshal11869) [inline src/UKHO.ERPFacade.API/Helpers/SapMessageBuilder.cs:38]:
```suggestion
        private const string EffectiveDateFormat = "yyyyMMdd";
        private const string EffectiveTimeFormat = "hhmmss";

```

File: src/UKHO.ERPFacade.Common/Services/ERPFacadeService.cs [hunk 2]
```diff
@@ -58,20 +58,21 @@ public List<UnitsOfSalePrices> MapAndBuildUnitsOfSalePrices(List<PriceInformatio
                             {
                                 var existingUnitOfSalePrice = unitsOfSalePriceList.Where(x => x.UnitName.Contains(priceInformation.ProductName)).FirstOrDefault();
 
-                                var effectiveUnitOfSalePriceDurations = existingUnitOfSalePrice.Price.Where(x => x.EffectiveDate.ToString("yyyyMMdd") == priceInformation.EffectiveDate).ToList();
+                                var effectiveUnitOfSalePriceDurations = existingUnitOfSalePrice!.Price.Where(x => x.EffectiveDate.ToString("yyyyMMdd") == priceInformation.EffectiveDate).ToList();
```
Comment (harshal11869) [inline src/UKHO.ERPFacade.Common/Services/ERPFacadeService.cs:61]: can we define the date format as below and use it wherever required ? 
private const string EffectiveDateFormat = "yyyyMMdd";

File: src/UKHO.ERPFacade.Common/Services/ERPFacadeService.cs [hunk 3]
```diff
@@ -58,20 +58,21 @@ public List<UnitsOfSalePrices> MapAndBuildUnitsOfSalePrices(List<PriceInformatio
...
 
                                 var futureUnitOfSalePriceDurations = existingUnitOfSalePrice.Price.Where(x => x.EffectiveDate.ToString("yyyyMMdd") == priceInformation.FutureDate).ToList();
                                 var futureStandard = futureUnitOfSalePriceDurations.Select(x => x.Standard).FirstOrDefault();
 
-                                if (effectiveStandard != null && !string.IsNullOrEmpty(priceInformation.EffectiveDate))
+                                if (effectiveStandard != null)
                                 {
-                                    if (!effectiveStandard.PriceDurations.Any(x => x.NumberOfMonths == Convert.ToInt32(priceInformation.Duration) && x.Rrp == Convert.ToDecimal(string.Format("{0:0.00}", Math.Round(Convert.ToDecimal(priceInformation.Price), 2)))))
+                                    if (!effectiveStandard.PriceDurations.Any(x => x.NumberOfMonths == Convert.ToInt32(priceInformation.Duration) && x.Rrp == Convert.ToDecimal(String.Format("{0:0.00}", Math.Round(Convert.ToDecimal(priceInformation.Price), 2)))))
```
Comment (harshal11869) [inline src/UKHO.ERPFacade.Common/Services/ERPFacadeService.cs:69]: Can we create method for getting decimal value of Rrp and use it wherever needed in this class ?

**Review by harshal11869 — CHANGES_REQUESTED** [review]

File: src/UKHO.ERPFacade.API/Helpers/SapMessageBuilder.cs (hunk 1 above)
Comment (sakshi15525) [reply src/UKHO.ERPFacade.API/Helpers/SapMessageBuilder.cs:38]: It is inline with the xml tag name.

**Review by jrippington — APPROVED** [review]
Looks good

**Review by JiviteshT — COMMENTED** [review]
Looks good

**Review by harshal11869 — APPROVED** [review]
Looks Good.

**Pull request merged**